app/blueprints/assessments/models/round_status.py

```diff
@@ -0,0 +1,48 @@
+from dataclasses import dataclass
+
+from app.blueprints.services.data_services import get_round
+from app.blueprints.services.models.round import Round
+from app.blueprints.shared.helpers import get_ttl_hash
+from config import Config
+from fsd_utils.simple_utils.date_utils import (
+    current_datetime_after_given_iso_string,
+)
+from fsd_utils.simple_utils.date_utils import (
+    current_datetime_before_given_iso_string,
+)
+
+
+@dataclass
+class RoundStatus:
+    is_application_not_yet_open: bool
+    is_application_open: bool
+    is_application_closed: bool
```

Comment:
Do both of these variables tell the same thing?  can we keep one and use `not` to check negative scenario?

Reply:
They kind of do, but not quite. Also they can both be `False` when `is_application_not_yet_open` is true, and having them as separate properties I think makes the logic easier to read in places rather than having additional `not` clauses.

Reply:
Actually I've renamed it now to `has_application_closed` so it's only `true` after the `deadline` has passed - which is useful for the link display thing you pointed out further down